mls-rs/src/group/snapshot.rs

```diff
@@ -23,83 +23,81 @@ use super::proposal_cache::{CachedProposal, ProposalCache};
-use mls_rs_core::identity::IdentityProvider;
+use mls_rs_core::{crypto::SignatureSecretKey, identity::IdentityProvider};
 
 #[cfg(all(feature = "std", feature = "by_ref_proposal"))]
 use std::collections::HashMap;
 
-#[cfg(all(feature = "by_ref_proposal", not(feature = "std")))]
 use alloc::vec::Vec;
 
-use super::{cipher_suite_provider, epoch::EpochSecrets, state_repo::GroupStateRepository};
+use super::{
+    cipher_suite_provider, epoch::EpochSecrets, state_repo::GroupStateRepository, ExportedTree,
+    TreeKemPublic,
+};
 
 #[derive(Debug, PartialEq, Clone, MlsEncode, MlsDecode, MlsSize)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub(crate) struct Snapshot {
-    version: u16,
-    state: RawGroupState,
-    private_tree: TreeKemPrivate,
-    epoch_secrets: EpochSecrets,
-    key_schedule: KeySchedule,
+#[non_exhaustive]
+pub struct Snapshot<'a> {
+    pub version: u16,
+    pub state: RawGroupState<'a>,
+    pub private_tree: TreeKemPrivate,
+    pub epoch_secrets: EpochSecrets,
+    pub key_schedule: KeySchedule,
     #[cfg(all(feature = "std", feature = "by_ref_proposal"))]
-    pending_updates: HashMap<HpkePublicKey, (HpkeSecretKey, Option<SignatureSecretKey>)>,
+    pub pending_updates: HashMap<HpkePublicKey, (HpkeSecretKey, Option<SignatureSecretKey>)>,
     #[cfg(all(not(feature = "std"), feature = "by_ref_proposal"))]
-    pending_updates: Vec<(HpkePublicKey, (HpkeSecretKey, Option<SignatureSecretKey>))>,
-    pending_commit: Option<CommitGeneration>,
-    signer: SignatureSecretKey,
+    pub pending_updates: Vec<(HpkePublicKey, (HpkeSecretKey, Option<SignatureSecretKey>))>,
+    pub pending_commit: Option<CommitGeneration>,
+    pub signer: SignatureSecretKey,
 }
 
-impl Snapshot {
+impl Snapshot<'_> {
     pub(crate) fn group_id(&self) -> &[u8] {
         &self.state.context.group_id
     }
 }
 
 #[derive(Debug, MlsEncode, MlsDecode, MlsSize, PartialEq, Clone)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub(crate) struct RawGroupState {
-    pub(crate) context: GroupContext,
+#[non_exhaustive]
+pub struct RawGroupState<'a> {
+    pub context: GroupContext,
     #[cfg(all(feature = "std", feature = "by_ref_proposal"))]
-    pub(crate) proposals: HashMap<ProposalRef, CachedProposal>,
+    pub proposals: HashMap<ProposalRef, CachedProposal>,
     #[cfg(all(not(feature = "std"), feature = "by_ref_proposal"))]
-    pub(crate) proposals: Vec<(ProposalRef, CachedProposal)>,
-    pub(crate) public_tree: TreeKemPublic,
-    pub(crate) interim_transcript_hash: InterimTranscriptHash,
-    pub(crate) pending_reinit: Option<ReInitProposal>,
-    pub(crate) confirmation_tag: ConfirmationTag,
+    pub proposals: Vec<(ProposalRef, CachedProposal)>,
+    pub public_tree: ExportedTree<'a>,
+    pub interim_transcript_hash: InterimTranscriptHash,
+    pub pending_reinit: Option<ReInitProposal>,
+    pub confirmation_tag: ConfirmationTag,
+    tree_internals: TreeKemPublic,
```

Comment:
This field ought to have a comment stating that its `nodes` field is likely empty and useless (except in one specific function where it gets repopulated).